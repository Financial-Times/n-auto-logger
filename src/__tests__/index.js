import logger, {
	logAction,
	logOperation,
	requestIdMiddleware,
	toMiddleware,
	tagService,
	enhancedRender,
	compose,
} from '../index';

describe('n-auto-logger exports', () => {
	it('logger', () => {
		expect(typeof logger).toBe('object');
	});

	it('logAction', () => {
		expect(typeof logAction).toBe('function');
	});

	it('logOperation', () => {
		expect(typeof logOperation).toBe('function');
	});

	it('toMiddleware', () => {
		expect(typeof toMiddleware).toBe('function');
	});

	it('requestIdMiddleware', () => {
		expect(typeof requestIdMiddleware).toBe('function');
	});

	it('tagService', () => {
		expect(typeof tagService).toBe('function');
	});

	it('enhancedRender', () => {
		expect(typeof enhancedRender).toBe('function');
	});

	it('compose', () => {
		expect(typeof compose).toBe('function');
	});
});
