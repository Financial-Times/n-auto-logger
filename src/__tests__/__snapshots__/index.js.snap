// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-auto-logger autoLog should invoke autoLogged async callFunction correctly 1`] = `
Array [
  Array [
    Object {
      "test": "a",
    },
    Object {
      "meta": "b",
    },
  ],
]
`;

exports[`n-auto-logger autoLog should invoke non-async callFunction correctly 1`] = `
Array [
  Array [
    Object {
      "test": "a",
    },
    Object {
      "meta": "b",
    },
  ],
]
`;

exports[`n-auto-logger autoLog should log async callFunction failure correctly and throw the original exception 1`] = `
Array [
  Array [
    Object {
      "action": "callFunction",
      "category": "EXCEPTION",
      "message": "bar",
      "meta": "b",
      "result": "failure",
      "test": "a",
    },
  ],
]
`;

exports[`n-auto-logger autoLog should log async callFunction success correctly 1`] = `
Array [
  Array [
    Object {
      "action": "callFunction",
      "meta": "b",
      "test": "a",
    },
  ],
  Array [
    Object {
      "action": "callFunction",
      "meta": "b",
      "result": "success",
      "test": "a",
    },
  ],
]
`;

exports[`n-auto-logger autoLog should log error if input arg meta is specified but not an Object 1`] = `"check the args format of autoLogged function [callFunction], it needs to be (params: Object, meta: Object), documents: https://github.com/Financial-Times/n-auto-logger/blob/master/README.md#function-args-format"`;

exports[`n-auto-logger autoLog should log error if input arg params is not an Object 1`] = `"check the args format of autoLogged function [callFunction], it needs to be (params: Object, meta: Object), documents: https://github.com/Financial-Times/n-auto-logger/blob/master/README.md#function-args-format"`;

exports[`n-auto-logger autoLog should log non-async callFunction success correctly 1`] = `
Array [
  Array [
    Object {
      "action": "callFunction",
      "meta": "b",
      "test": "a",
    },
  ],
  Array [
    Object {
      "action": "callFunction",
      "meta": "b",
      "result": "success",
      "test": "a",
    },
  ],
]
`;

exports[`n-auto-logger autoLog should throw error if there're more than 2 args 1`] = `"check the args format of autoLogged function [callFunction], it needs to be (params: Object, meta: Object), documents: https://github.com/Financial-Times/n-auto-logger/blob/master/README.md#function-args-format"`;
